nav2_dynamic_params/include/nav2_dynamic_params/dynamic_params_validator.hpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_DYNAMIC_PARAMS__DYNAMIC_PARAMS_VALIDATOR_HPP_
+#define NAV2_DYNAMIC_PARAMS__DYNAMIC_PARAMS_VALIDATOR_HPP_
+
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+#include "rclcpp/rclcpp.hpp"
+
+namespace nav2_dynamic_params
+{
+
+class DynamicParamsValidator
+{
+public:
+  explicit DynamicParamsValidator(rclcpp::Node::SharedPtr node);
+
+  ~DynamicParamsValidator() {}
+
+  void add_param(std::string param_name, rclcpp::ParameterType type);
```

Comment:
looks like some of the parameters in these functions could be passed as const refs.